xonsh/prompt/vc.py

```diff
@@ -20,9 +20,14 @@
     "RE_REMOVE_ANSI",
 )
 
+# when running git status commands we do not want to acquire locks running command like git status
+_GIT_OPTIONAL_LOCKS = {"GIT_OPTIONAL_LOCKS": "0"}
```

Comment:
Can you remove this as a global variable? Many people will never touch this if they are not using `curr_branch`